New Playbook: dispatch_playbooks

kelby-shelton opened the pull request:
# New playbook called dispatch_playbooks

## Description
An alternative to the built-in playbook launcher that allows a user to launch multiple playbooks in one block based on dynamic data.

## Features
- Collects indicator data from event and routes it to specified input style playbooks.
- Attempting to launch community input playbooks will cause error.
- Waits for child playbooks to finish and collects their outputs and passes them forward.

kelby-shelton committed:
New Playbook: dispatch_playbooks

# New playbook called dispatch_playbooks

## Description
An alternative to the built-in playbook launcher that allows a user to launch multiple playbooks in one block based on dynamic data.

## Features
- Collects indicator data from event and routes it to specified input style playbooks.
- Attempting to launch community input playbooks will cause error.
- Waits for child playbooks to finish and collects their outputs and passes them forward.

philroyer-phantom commented on dispatch_playbooks.py:
```diff
@@ -0,0 +1,406 @@
+        "indicator_types_exclude": None,
+        "indicator_tags_include": ', '.join([item[0] for item in indicator_tags_include if item]),
+        "indicator_tags_exclude": ', '.join([item[0] for item in indicator_tags_exclude if item]),
+        "artifact_ids_include": playbook_input_artifact_ids_include_values,
```
I am seeing an exception in the CF here if we pass in [""] when this playbook is called with no artifact IDs. How about changing this line to the following:
`        "artifact_ids_include": ', '.join([item[0] for item in playbook_input_artifact_ids_include_values if item]),`
?

philroyer-phantom reviewed (commented):
See line 335.

kelby-shelton committed:
Multiple changes to outputs

- Added support for sub_playbook_outputs
- Added support for sub_playbook_inputs
- Fixed artifact_ids_include error
- Added new output variables with better descriptions.

kelby-shelton committed:
Better defaults control

- Checks for at least 1 playbook tag
- Errors out if at least 1 playbook tag not found (works with empty strings)
- Defaults to local if no repos specified (works with empty strings)

kelby-shelton committed:
Renamed to dispatch_input_playbooks

philroyer-phantom commented:
Looks like there are two files left over from before the rename. Could you remove dispatch_playbooks.json and dispatch_playbooks.png please?

kelby-shelton committed:
Removed old files

philroyer-phantom commented on dispatch_input_playbooks.json:
```diff
@@ -0,0 +1,614 @@
+                    "y": 474
+                }
+            },
+            "notes": "- Must have at least one input style playbook in whichever repo you choose for this utility to work.\n- Playbook will error out if community repo is provided to ensure that users do not accidentally launch community input playbooks that have not been configured."
```
Please add a note that artifact scope will be ignored. That can be a bit surprising for someone not used to how the "indicator_collect" function works.

kelby-shelton replied:
Done

kelby-shelton committed:
Updated Notes section

Added note about artifact scope being ignored.

kelby-shelton committed:
Updated input descriptions
